### ISSUE TITLE
Deletion of named constraints and creation of identically named constraints

### ISSUE THREAD

**Issue by Marthioo** [jump-dev/JuMP.jl#2257]
Hello Github! I have been trying to implement some code, the general idea can be seen below. I have two types of constraints, depending on the feasibility of some subproblems I can get either a constraint of type (1, opt cut) or type (2, feas cut). What I want to accomplish is that if all subproblems are optimal, then I add a constraint of type (1) for every s in S. But if a single subproblem is infeasible, resulting in a constraint of type (2), then I want to delete all of the constraints of type (1) added to the model in that same loop, while adding the single constraint of type (2). 
The problem I run into is with naming the constraints and referencing the correct ones when deleting. Currently I have tried how I thought it should work, but once I get a constraint of type (2) it cannot find the constraints of type (1) to delete.

![image](https://user-images.githubusercontent.com/66962612/84683100-f911a780-af36-11ea-8e0b-0c468b5c5d2c.png)

I have tried a few different approaches like saving the constraints with @build_constraint and then adding the constraints after the loop, but again I ran into a problem since I could not figure out how to define a vector of constraints. I tried using the below but obviously it does not work, if anyone knows how to initialize a vector of constraints which I can add constraints to, then I think my problem would be solved as well.
![image](https://user-images.githubusercontent.com/66962612/84683724-efd50a80-af37-11ea-9165-bb37ad476013.png)

Thank you in advance!

**Comment by odow** [comment]
You don't have to use JuMP's built-in containers. Just use a dictionary.
```Julia
optcut = Dict()
if opt == 1
    optcut[s] = @constraint(...)
else
    for (ss, con) in optcut
        delete(model, con)
    end
end
```

You should also read https://github.com/jump-dev/JuMP.jl/issues/1956 to understand the issues with deleting constraints in a built-in JuMP container.

Finally, these questions are better asked on the forum: https://discourse.julialang.org/c/domain/opt/13
You should also read the first post of https://discourse.julialang.org/t/psa-make-it-easier-to-help-you/14757/19. In particular, don't post screenshots. Provide the text instead which makes it searchable and allows people to copy-paste your code into their terminal.

**Comment by Marthioo** [comment]
Thank you for the answer. I tried to read #1956 before posting but I could not figure out how to apply it to my problem. 
I am very sorry for posting in the wrong place and uploading screenshots instead of text, I'm new to this site, going forward I will post in the correct way :)  
I could not get it to work with your snippet of code either, the optcuts get defined but when I try to delete them it cannot find them. 
I have tried a different approach now, if I need help I will post in the forums instead. Sorry!
